flink-runtime/src/main/java/org/apache/flink/runtime/state/StateSnapshotRestore.java

```diff
@@ -36,8 +39,10 @@
      * hint.
      *
      * @param readVersionHint the required version of the state to read.
+     * @param dataInputStream the current dataInputStream.
      * @return a reader that reads state by key-groups, for the given read version.
      */
     @Nonnull
-    StateSnapshotKeyGroupReader keyGroupReader(int readVersionHint);
+    StateSnapshotKeyGroupReader keyGroupReader(int readVersionHint, DataInputStream dataInputStream)
```

Comment:
Could you please explain why adding `dataInputStream` could distinguish between full/inc snapshots on recovery?

Moreover, dose `StateSnapshotKeyGroupReader#readMappingsInKeyGroup(@Nonnull DataInputView div, @Nonnegative int keyGroupId)` still need that `div` ?

Reply:
You are right, `inputView` can be passed once to `readMappingsInKeyGroup`.
I reverted the addition of parameter and increased the visibility of `StateTableByKeyGroupReaders`.